### PR TITLE
add more tips to udf docs

### PR DESCRIPTION
**Pull request by jdries**

### CODE REVIEW

File: docs/udf.rst [hunk 1]
```diff
@@ -78,6 +78,18 @@ in smaller chunks and process these chunks separately (e.g. on isolated workers)
 Consequently, it's important that your **UDF algorithm operates correctly
 in such a chunked processing context**.
 
+A very common mistake is to use index-based array indexing, rather than name based. The index based approach
+assumes that datacube dimension order is fixed, which is not guaranteed. Next to that, it also reduces the readability
+of your code. Label based indexing is a great feature of xarray, and should be used whenever possible.
```
Comment (soxofaan) [inline docs/udf.rst:83]: I think it's worth to put a more concrete example here

File: docs/udf.rst [hunk 2]
```diff
@@ -78,6 +78,18 @@ in smaller chunks and process these chunks separately (e.g. on isolated workers)
...
+
+As a rule of thumb, the UDF should preserve the dimensions, their order, and shape of the input
```
Comment (soxofaan) [inline docs/udf.rst:85]: I think users should not worry about dimension order (as long as they use something like Xarray), so I would not mention "should preserve ... their order" here

Comment (jdries) [reply docs/udf.rst]: remove 'their order'

**Comment by jdries** [comment]
@soxofaan  addressed one comment, adding an example takes more time. I would merge this important update now, so that users at least have some extra guidance.

**Pull request merged**